const_safety.md

```diff
@@ -1 +1,134 @@
-# Const safety
\ No newline at end of file
+# Const safety
+
+The miri engine, which is used to execute code at compile time, can fail in
+four possible ways:
+
+* The program performs an unsupported operation (e.g., calling an unimplemented
+  intrinsics, or doing an operation that would observe the integer address of a
+  pointer).
+* The program causes undefined behavior (e.g., dereferencing an out-of-bounds
+  pointer).
+* The program panics (e.g., a failed bounds check).
+* The program loops forever, and this is detected by the loop detector.
```

Comment:
On a best effort basis. We basically don't care too much about nontermination except for diagnostics

Reply:
AFAIK we do error hard if we detect a reoccurring state? This could happen in a promoted as well.